bin/util.py

```diff
@@ -643,24 +643,20 @@ def write_yaml(data, path=None, allow_yamllib=False):
 
 
 # Parse validation mode
-def parse_validation(mode):
+def parse_validation(mode: str) -> set[str]:
     if mode == "default":
-        return {
-            "mode": "default",
-            "interactive": False,
-            "multi-pass": False,
-        }
+        return {mode}
     else:
-        res = {
-            "mode": "custom",
-            "interactive": "interactive" in mode,
-            "multi-pass": "multi-pass" in mode,
-        }
-        temp = mode.replace(" interactive", "", 1)
-        temp = temp.replace(" multi-pass", "", 1)
-        if temp != "custom":
+        ok = True
+        parsed = set()
+        for part in mode.split():
+            if part in ["custom", "interactive", "multi-pass"] and part not in parsed:
+                parsed.add(part)
+            else:
+                ok = False
+        if "custom" not in parsed or not ok:
             fatal(f"Unrecognised validation mode {mode}.")
-        return res
+        return parsed
```

Comment:
I think this is moving towards https://icpc.io/problem-package-format/spec/2023-07-draft.html#type, which is good. However, `default` no longer exists in `2023-07-draft`, I think this is now `pass-fail`?

Also, `custom` appears to be gone, but maybe that's simply determined based on the existence of an `output_validator/`? :thinking: (that appears to be the case, based on https://icpc.io/problem-package-format/spec/2023-07-draft.html#overview-1)

Reply:
yeah this is the legacy parser i dont want to touch the new type yet

Reply:
Fair enough :stuck_out_tongue: I started an attempt, but it's a bit more complicated than I thought. Since this PR is already growing quite large, perhaps it's better to simply merge this PR, and then I'll open a new one to use the new `type`?